src/data/reducers/filters.js

```diff
@@ -0,0 +1,39 @@
+import { GOT_GRADES, FILTER_BY_ASSIGNMENT_TYPE } from '../constants/actionTypes/grades';
+
+import { INITIALIZE_FILTERS, UPDATE_ASSIGNMENT_FILTER } from '../constants/actionTypes/filters';
+
+const initialState = {};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case FILTER_BY_ASSIGNMENT_TYPE:
+      return {
+        ...state,
+        assignmentType: action.filterType,
+        assignment: (
+          action.filterType !== 'All' &&
+          (state.assignment || {}).type !== action.filterType)
+          ? 'All' : state.assignment,
+      };
+    case INITIALIZE_FILTERS:
+      return {
+        ...state,
+        ...action.data,
+      };
+    case GOT_GRADES:
```

Comment:
If I'm understanding correctly, one of the bugs you mentioned involves the fact that if you load the page with query parameters dictating the assignment filter, you don't yet have access to the subsection metadata, like the type and the id, so the state doesn't include that information when the page loads. In that case, could you get that information and add it to the state when the GOT_GRADES action is being handled?

Reply:
Be careful what you wish for.... https://github.com/edx/gradebook/pull/108/files#diff-0b7c15d7052c2ebb52aa2bb06d04566dR28